packages/doenetml-worker-rust/lib-doenetml-core/src/core/general_state_var/mirror_state_var.rs

```diff
@@ -0,0 +1,71 @@
+#[cfg(test)]
+#[path = "mirror_state_var.test.rs"]
+mod tests;
+
+use crate::components::prelude::*;
+
+/// A struct of all data required to compute the value of this state variable.
+#[add_dependency_data]
+#[derive(Debug, Default, StateVariableDependencies, StateVariableDataQueries)]
+pub struct MirrorRequiredData<T>
+where
+    T: Default + Clone,
+{
+    mirrored_value: StateVarView<T>,
+}
+
+/// A state variable that mirrors another state variable from the same component.
+///
+/// The state variable has a single data query, which is for the value of another state variable.
+/// It takes on the value of this other state variable,
+/// and when inverting, requests that the value of this other variable be changed.
+#[derive(Debug, Default)]
+pub struct MirrorStateVar {
+    mirrored_state_var_idx: StateVarIdx,
+}
+
+impl MirrorStateVar {
+    pub fn new(mirrored_state_var_idx: StateVarIdx) -> Self {
```

Comment:
I'm torn here between following Rust's convention and our own. Rust convention is to have `new` take no arguments. So this would be called something like `with_index` or `from_index`

Reply:
It seems like `new` with arguments is OK:

From https://rust-lang.github.io/api-guidelines/predictability.html:
> The name new should generally be used for the primary method of instantiating a type. Sometimes it takes no arguments, as in the examples above. Sometimes it does take arguments, like [Box::new](https://doc.rust-lang.org/stable/std/boxed/struct.Box.html#method.new) which is passed the value to place in the Box.